zookeeper/settings.sls

```diff
@@ -67,14 +67,38 @@
 {%- set hosts_target         = g.get('hosts_target', p.get('hosts_target', 'roles:zookeeper')) %}
 {%- set targeting_method     = g.get('targeting_method', p.get('targeting_method', 'grain')) %}
 
+{%- set cluster_id     = g.get('cluster_id', p.get('cluster_id', None)) %}
 {%- set zookeepers_with_ids = [] %}
 {%- set zookeepers          = [] %}
-{%- set myid_dist           = [] %}
+{%- set myid_dist            = [] %}
 {%- set connection_string   = [] %}
-{%- set minion_ips          = salt['network.ip_addrs']() %}
+{%- set minion_ids          = [salt['network.get_hostname'](),
+                               grains['id'],
+                               grains['fqdn'],
+                               grains['nodename']] + salt['network.ip_addrs']() %}
 
 {% if p.get('nodes') %}
   {%- set zookeeper_nodes   = p.get('nodes', []) %}
+{% elif p.get('clusters') and cluster_id != None %}
+  {%- set zookeeper_nodes   = p.get('clusters').get(cluster_id, []) %}
+{% elif p.get('clusters') %}
+  {%- set zookeeper_nodes_tmp   = [] %}
+  {%- if p.get('clusters')|length == 1 %}
+    {%- set zookeeper_nodes   = p.get('clusters').values()[0] %}
+  {%- else %}
+    {% for cluster, nodes in p.get('clusters').iteritems() %}
+      {%- for node in nodes %}
+        {%- if node in minion_ids %}
+          {%- do zookeeper_nodes_tmp.append(nodes)  %}
+          {%- break %}
+        {%- endif %}
+      {%- endfor %}
+      {%- if zookeeper_nodes_tmp|length != 0 %} 
+        {%- break %}
+      {%- endif %}
+    {% endfor %}
+    {%- set zookeeper_nodes   = zookeeper_nodes_tmp[0] if zookeeper_nodes_tmp|length !=0 else [] %}  
+  {%- endif %}
```

Comment:
Thanks for explanation. But I still don't understand why do we need to sort our dictionary of clusters? Isn't it better to check only values, for example like this?:
`{%- for nodes in p.clusters.values() %}`
I mean, it doesn't matter if our clusters are checked not in order. 
If `values()` has also problems on some versions of python then we can use `dictsort`.